### ISSUE TITLE
Execution events listener

### ISSUE THREAD

**Issue by DavidGalvis** [SeleniumHQ/selenium-ide#514]
## 🚀 Feature Proposal

to be able to pass in the command line a file listener path that allows us to share the events that occur during the execution and obtain the most relevant values.

The events to listen to can be:

startTest
endTest
startStep
endStep
endSuite

That each event can return its result and the test case in which it happened.

## Motivation

The objective is that the selenium-side-runner can be integrated into other node projects and personalized actions can be performed depending on what happens in the execution.

## Example

listeners = {
    'startTest': function(test, result) {
      console.log("Listener: test run starting!");
      console.log("Listener: result: " +result);
    },
    'endTest': function(test, result) {
      console.log("Listener: test run ending!");
      console.log("Listener: result: " +result);
    },
    'startStep': function(test, step, result) {
      console.log("Listener: step starting!");
      console.log("Listener: result: " +result);
    },
    'endStep': function(test, step, result) {
      console.log("Listener: step ending!");
      console.log("Listener: result: " +result);
    },
    'endSuite': function(testSuite, result) {
      console.log("Listener: all test ended!");
      console.log("Listener: result: " +result);
    }
}

**Comment by corevo** [comment]
That's a great idea, currently the IDE's test runner, doesn't allow for each step events, but we can certainly do that with `beforeAll`, `beforeEach`, `afterAll` and `afterEach`.  
I'm not sure about the possibility of passing that information over, as it is not given to me by [jest](https://jestjs.io/docs/en/api#afterallfn-timeout).  
Would it still be useful without it?

**Comment by DavidGalvis** [comment]
Yes, it would still be useful. How can I use beforeAll, beforeEach, afterAll and afterEach in Selenium-side-runner?
